toolchain/sem_ir/absolute_node_id.cpp

```diff
@@ -95,7 +95,7 @@ static auto GetAbsoluteNodeIdImpl(
     }
 
     // If the parse node has a value, use it for the location.
-    if (auto loc_id = cursor_ir->insts().GetLocId(cursor_inst_id);
+    if (auto loc_id = cursor_ir->insts().GetResolvedLocId(cursor_inst_id);
```

Comment:
Here you're changing both calls to `HandleLocId`; had you considered passing `cursor_inst_id` in as the `loc_id` argument and having `HandleLocId` do the `GetResolvedLocId` call for both callers?

Reply:
We only want to call it here if the canonical locid has a value, though, and we only continue if it returns false but did have a value. I think we'd need a tristate return from HandleLocId then.